lms/djangoapps/discussion_api/pagination.py

```diff
@@ -35,33 +35,52 @@ def previous_page_number(self):
         return self.page_num - 1
 
 
-def get_paginated_data(request, results, page_num, per_page):
+class DiscussionAPIPagination(NamespacedPageNumberPagination):
     """
-    Return a dict with the following values:
-
-    next: The URL for the next page
-    previous: The URL for the previous page
-    results: The results on this page
+    Subclasses NamespacedPageNumberPagination to provide custom implementation of pagination metadata
+    by overriding it's methods
     """
-    # Note: Previous versions of this function used Django Rest Framework's
-    # paginated serializer.  With the upgrade to DRF 3.1, paginated serializers
-    # have been removed.  We *could* use DRF's paginator classes, but there are
-    # some slight differences between how DRF does pagination and how we're doing
-    # pagination here.  (For example, we respond with a next_url param even if
-    # there is only one result on the current page.)  To maintain backwards
-    # compatability, we simulate the behavior that DRF used to provide.
-    page = _Page(results, page_num, per_page)
-    next_url, previous_url = None, None
-    base_url = request.build_absolute_uri()
+    def __init__(self, request, page_num, num_pages):
+        """
+        Overrides parent constructor to take information from discussion api
+        essential for the parent method
+        """
+        self.page = _Page(page_num, num_pages)
+        self.base_url = request.build_absolute_uri()
+        self.num_pages = num_pages
+
+        super(DiscussionAPIPagination, self).__init__()
+
+    def get_result_count(self):
+        """
+        A count can't be calculated with the information coming from the
+        Forum's api, so returning 0 for the parent method to work
+        """
+        # TODO: return actual count
+        return 0
```

Comment:
@jcdyer I forgot to point out that @wajeeha-khalid is already working on adding the count to the comments service API so that we can return it from here. We will have those changes merged into this branch before we merge to master. So we do not have worry about this.